pybamm/solvers/solution.py

```diff
@@ -780,6 +780,13 @@ def copy(self):
 
         return new_sol
 
+    def condition_number(self, model):
+        # returns condition number for the solution
+        f = model.rhs_algebraic_eval(self.t[0], self.y[:, 0], [])
+        J = model.jac_rhs_algebraic_eval(self.t[0], self.y[:, 0], [])
```

Comment:
loop over all indices of `self.t` and `self.y` and return a vector of length `len(sol.t)`